### ISSUE TITLE
Layer build error for python3.9

### ISSUE THREAD

**Issue by MrAtheist** [terraform-aws-modules/terraform-aws-lambda#313]
## Description

Getting `entrypoint requires the handler name to be the first argument` while trying to build the lambda layer via docker with python3.9 as runtime. Weird enough it works with python3.8, so im not sure whats missing here...

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 3.2.0
- Terraform version: v0.15.1
- MacOS: 12.1 running M1

- Provider version(s):
```
- Installed hashicorp/null v3.1.1 (signed by HashiCorp)
- Installed hashicorp/aws v4.14.0 (signed by HashiCorp)
- Installed hashicorp/archive v2.2.0 (signed by HashiCorp)
- Installed hashicorp/external v2.2.2 (signed by HashiCorp)
- Installed hashicorp/local v2.2.2 (signed by HashiCorp)
```

## Reproduction Code [Required]

```
module "lambda_layer" {
  source = "terraform-aws-modules/lambda/aws"
  version = "3.2.0"

  create_function     = false
  create_layer        = true
  layer_name          = "helloworld-layer"
  compatible_runtimes = ["python3.9"]
  runtime             = "python3.9"

  source_path = [
    {
      path             = "${path.module}/src"
      pip_requirements = true
      prefix_in_zip    = "python" # required to get the path correct
    }
  ]

  build_in_docker = true
  docker_image    = "public.ecr.aws/sam/build-python3.9"
  docker_file     = "${path.module}/src/Dockerfile"
}
```

```
# ${path.module}/src/Dockerfile

FROM public.ecr.aws/lambda/python:3.9

COPY requirements.txt  .
RUN  pip3 install -r requirements.txt --target "${LAMBDA_TASK_ROOT}"

CMD ["index.handler"]
```

```
terraform init
terraform plan
terraform apply --auto-approve
```

## Expected behavior

docker run should behave correctly

```
docker run --rm -w /var/task
-v /private/var/folders/dd/fvl7_qz94f34p57zmsgzycth0000gn/T/terraform-aws-lambda-6u_93uta:/var/task:z
-v /Users/mratheist/.ssh/known_hosts:/root/.ssh/known_hosts:z 494788200de1 
/bin/sh -c 'python3.9 -m pip install --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R 501:20 .'
```
### Terminal Output Screenshot(s)

```
# folder dir is loosely obfuscated... 

Error: Error running command 'builds/4596cd1b10d43761d40b1f13167df4a0f06830146de508936b3da4e345ef0c38.plan.json': exit status 1. Output: zip: creating 'builds/4596cd1b10d43761d40b1f13167df4a0f06830146de508936b3da4e345ef0c38.zip' archive
│ > docker images '--format={{.ID}}' public.ecr.aws/sam/build-python3.9
│ Installing python requirements: ../../modules/lambda-decrypt-kinesis/src/requirements.txt
│ > mktemp -d terraform-aws-lambda-XXXXXXXX # /var/folders/dd/fvl7_qz94f34p57zmsgzycth0000gn/T/terraform-aws-lambda-6u_93uta
│ > cd /var/folders/dd/fvl7_qz94f34p57zmsgzycth0000gn/T/terraform-aws-lambda-6u_93uta
│ > docker run --rm -w /var/task -v /private/var/folders/dd/fvl7_qz94f34p57zmsgzycth0000gn/T/terraform-aws-lambda-6u_93uta:/var/task:z -v /Users/mratheist/.ssh/known_hosts:/root/.ssh/known_hosts:z 494788200de1 /bin/sh -c 'python3.9 -m pip install --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R 501:20 .'
│ entrypoint requires the handler name to be the first argument
│ zip: Error during zip archive creation
│ Traceback (most recent call last):
│   File ".terraform/modules/lambda_decrypt_kinesis.lambda_layer/package.py", line 1277, in build_command
│     bpm.execute(build_plan, zs, query)
│   File ".terraform/modules/lambda_decrypt_kinesis.lambda_layer/package.py", line 811, in execute
│     with install_pip_requirements(query, pip_requirements, tmp_dir) as rd:
│   File "/opt/homebrew/Cellar/python@3.9/3.9.12_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py", line 119, in __enter__
│     return next(self.gen)
│   File ".terraform/modules/lambda_decrypt_kinesis.lambda_layer/package.py", line 948, in install_pip_requirements
│     check_call(docker_run_command(
│   File "/opt/homebrew/Cellar/python@3.9/3.9.12_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
│     raise CalledProcessError(retcode, cmd)
│ subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-w', '/var/task', '-v', '/private/var/folders/dd/fvl7_qz94f34p57zmsgzycth0000gn/T/terraform-aws-lambda-6u_93uta:/var/task:z', '-v', '/Users/mratheist/.ssh/known_hosts:/root/.ssh/known_hosts:z', '494788200de1', '/bin/sh', '-c', 'python3.9 -m pip install --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R 501:20 .']' returned non-zero exit status 142.
```

**Comment by micah5** [comment]
I have the same issue on 3.8. Only shows up in CI process.. still struggling to debug why

**Comment by micah5** [comment]
So, as far as I can tell`public.ecr.aws/sam/build-python` is the culprit. I've switched to a community run repo [here](https://gallery.ecr.aws/n8r6f1x4/lambci-temporary) on public ECR that uses lambci directly and it seems to be working now.

**Comment by antonbabenko** [comment]
This is still an issue, and #316 does not seem to fix it. Updates are highly appreciated.

**Comment by antonbabenko** [comment]
To reproduce it:
```
cd examples/build-package
terraform init
terraform apply -target="module.lambda_layer"
```

Output:
```
module.lambda_layer.null_resource.archive[0]: Creating...
module.lambda_layer.null_resource.archive[0]: Provisioning with 'local-exec'...
module.lambda_layer.null_resource.archive[0] (local-exec): Executing: ["python3" "../../package.py" "build" "--timestamp" "1662026254442858000" "builds/8e6338d3e2d57de107a95f1f99973201ab2964bb45c98a63db3de3796bf67890.plan.json"]
module.lambda_layer.null_resource.archive[0] (local-exec): zip: creating 'builds/8e6338d3e2d57de107a95f1f99973201ab2964bb45c98a63db3de3796bf67890.zip' archive
module.lambda_layer.null_resource.archive[0] (local-exec): > docker images '--format={{.ID}}' public.ecr.aws/sam/build-python3.8
module.lambda_layer.null_resource.archive[0] (local-exec): Installing python requirements: ./../fixtures/python3.8-app1/requirements.txt
module.lambda_layer.null_resource.archive[0] (local-exec): > mktemp -d terraform-aws-lambda-XXXXXXXX # /var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw
module.lambda_layer.null_resource.archive[0] (local-exec): > cd /var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw
module.lambda_layer.null_resource.archive[0] (local-exec): > docker run --rm -w /var/task -v /private/var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw:/var/task:z -v /Users/Bob/.ssh/known_hosts:/root/.ssh/known_hosts:z --entrypoint '' '9453fa734ff9
module.lambda_layer.null_resource.archive[0] (local-exec): eb960e2fbbf5' /bin/sh -c 'python3.8 -m pip install --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R 501:20 .'
module.lambda_layer.null_resource.archive[0] (local-exec): docker: invalid reference format.
module.lambda_layer.null_resource.archive[0] (local-exec): See 'docker run --help'.
module.lambda_layer.null_resource.archive[0] (local-exec): zip: Error during zip archive creation
module.lambda_layer.null_resource.archive[0] (local-exec): Traceback (most recent call last):
module.lambda_layer.null_resource.archive[0] (local-exec):   File "/Users/Bob/Sites/terraform-aws-modules/terraform-aws-lambda/examples/build-package/../../package.py", line 1279, in build_command
module.lambda_layer.null_resource.archive[0] (local-exec):     bpm.execute(build_plan, zs, query)
module.lambda_layer.null_resource.archive[0] (local-exec):   File "/Users/Bob/Sites/terraform-aws-modules/terraform-aws-lambda/examples/build-package/../../package.py", line 811, in execute
module.lambda_layer.null_resource.archive[0] (local-exec):     with install_pip_requirements(query, pip_requirements, tmp_dir) as rd:
module.lambda_layer.null_resource.archive[0] (local-exec):   File "/opt/homebrew/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/contextlib.py", line 135, in __enter__
module.lambda_layer.null_resource.archive[0] (local-exec):     return next(self.gen)
module.lambda_layer.null_resource.archive[0] (local-exec):   File "/Users/Bob/Sites/terraform-aws-modules/terraform-aws-lambda/examples/build-package/../../package.py", line 948, in install_pip_requirements
module.lambda_layer.null_resource.archive[0] (local-exec):     check_call(docker_run_command(
module.lambda_layer.null_resource.archive[0] (local-exec):   File "/opt/homebrew/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 369, in check_call
module.lambda_layer.null_resource.archive[0] (local-exec):     raise CalledProcessError(retcode, cmd)
module.lambda_layer.null_resource.archive[0] (local-exec): subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-w', '/var/task', '-v', '/private/var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw:/var/task:z', '-v', '/Users/Bob/.ssh/known_hosts:/root/.ssh/known_hosts:z', '--entrypoint', '', '9453fa734ff9\neb960e2fbbf5', '/bin/sh', '-c', 'python3.8 -m pip install --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R 501:20 .']' returned non-zero exit status 125.
╷
│ Warning: Applied changes may be incomplete
│ 
│ The plan was created with the -target option in effect, so some changes requested
│ in the configuration may have been ignored and the output values may not be fully
│ updated. Run the following command to verify that no other changes are pending:
│     terraform plan
│       
│ Note that the -target option is not suitable for routine use, and is provided
│ only for exceptional situations such as recovering from errors or mistakes, or
│ when Terraform specifically suggests to use it as part of an error message.
╵
╷
│ Error: local-exec provisioner error
│ 
│   with module.lambda_layer.null_resource.archive[0],
│   on ../../package.tf line 65, in resource "null_resource" "archive":
│   65:   provisioner "local-exec" {
│ 
│ Error running command
│ 'builds/8e6338d3e2d57de107a95f1f99973201ab2964bb45c98a63db3de3796bf67890.plan.json':
│ exit status 1. Output: zip: creating
│ 'builds/8e6338d3e2d57de107a95f1f99973201ab2964bb45c98a63db3de3796bf67890.zip'
│ archive
│ > docker images '--format={{.ID}}' public.ecr.aws/sam/build-python3.8
│ Installing python requirements: ./../fixtures/python3.8-app1/requirements.txt
│ > mktemp -d terraform-aws-lambda-XXXXXXXX #
│ /var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw
│ > cd
│ /var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw
│ > docker run --rm -w /var/task -v
│ /private/var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw:/var/task:z
│ -v /Users/Bob/.ssh/known_hosts:/root/.ssh/known_hosts:z --entrypoint ''
│ '9453fa734ff9
│ eb960e2fbbf5' /bin/sh -c 'python3.8 -m pip install --no-compile --prefix=
│ --target=. --requirement=requirements.txt && chown -R 501:20 .'
│ docker: invalid reference format.
│ See 'docker run --help'.
│ zip: Error during zip archive creation
│ Traceback (most recent call last):
│   File "/Users/Bob/Sites/terraform-aws-modules/terraform-aws-lambda/examples/build-package/../../package.py", line 1279, in build_command
│     bpm.execute(build_plan, zs, query)
│   File "/Users/Bob/Sites/terraform-aws-modules/terraform-aws-lambda/examples/build-package/../../package.py", line 811, in execute
│     with install_pip_requirements(query, pip_requirements, tmp_dir) as rd:
│   File "/opt/homebrew/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/contextlib.py", line 135, in __enter__
│     return next(self.gen)
│   File "/Users/Bob/Sites/terraform-aws-modules/terraform-aws-lambda/examples/build-package/../../package.py", line 948, in install_pip_requirements
│     check_call(docker_run_command(
│   File "/opt/homebrew/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 369, in check_call
│     raise CalledProcessError(retcode, cmd)
│ subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-w',
│ '/var/task', '-v',
│ '/private/var/folders/vy/wm1qhfns0rl564mb6w5w1b3c0000gn/T/terraform-aws-lambda-3gysq0pw:/var/task:z',
│ '-v', '/Users/Bob/.ssh/known_hosts:/root/.ssh/known_hosts:z', '--entrypoint', '',
│ '9453fa734ff9\neb960e2fbbf5', '/bin/sh', '-c', 'python3.8 -m pip install
│ --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R
│ 501:20 .']' returned non-zero exit status 125.
```

**Comment by deybhayden** [comment]
Hey all, I was bit by this as well this afternoon. #316 didn't fix this issue because the PR was closed - not merged. I think if it was merged it'd take care of the issue for us - I verified with a manual `docker run locally` that I was able to get around the error:

```
docker run --entrypoint "" --rm -w /var/task -v /private/var/folders/14/asdfasdfasdf/T/terraform-aws-lambda-123:/var/task:z -v /Users/foo/.ssh/known_hosts:/root/.ssh/known_hosts:z 6eb3e2f3cf6a /bin/sh -c python3.9 -m pip install --no-compile --prefix= --target=. --requirement=requirements.txt && chown -R 501:20 .
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.0.2 :tada:
